src/TS_HW4/hw4.ts

```diff
@@ -0,0 +1,136 @@
+//  1. Створіть інтерфейс, який описує структуру об'єкта,що представляє калькулятор.
+//  Калькулятор повинен мати методи для виконання арифметичних операцій: додавання, віднімання, множення та ділення.
+//  Потім створіть функцію calculate, яка приймає об'єкт цього типу та виконує операцію і повертає результат.
+
+enum Actions {
+  Addition,
+  Subtraction,
+  Multiplication,
+  Division,
+}
+
+interface ICalculator {
+  addition(firstNumber: number, secondNumber: number): number;
+  subtraction(firstNumber: number, secondNumber: number): number;
+  multiplication(firstNumber: number, secondNumber: number): number;
+  division(firstNumber: number, secondNumber: number): number;
+}
+
+class Calculator implements ICalculator {
+  addition(firstNumber: number, secondNumber: number): number {
+    return firstNumber + secondNumber;
+  }
+
+  subtraction(firstNumber: number, secondNumber: number): number {
+    return firstNumber - secondNumber;
+  }
+
+  multiplication(firstNumber: number, secondNumber: number): number {
+    return firstNumber * secondNumber;
+  }
+
+  division(firstNumber: number, secondNumber: number): number {
+    return firstNumber / secondNumber;
```

Comment:
If you're up for a challenge, why not try complicating the task by adding a case where the user tries to divide by zero? It'll definitely make things more interesting! :-)

Reply:
ok ) 

  division(firstNumber: number, secondNumber: number): number {
    if (secondNumber === 0) throw new EvalError('Division by zero is prohibited!');
    return firstNumber / secondNumber;